### PR TITLE
Facet Styling

### LINKED ISSUE #803: Add additional "rule" marks when COLUMN/ROW is mapped, but X/Y is not
For example, to complete text table's table border.  
vega_editor

(Whether we add table border by default is orthogonal -- if the borders are added, it should be completed.

### LINKED ISSUE #920: Draw x,y grid inside facet
Current faceted charts have shared grid across multiple like this:

vega_editor

However, having separate grid for different singles like an example below is nicer.

——

## TODO
- [ ] add a property for `split` axis mode (Need to think what should be the name though)
  - In this mode
    - [ ] hide grid from the outside shared x/y axis
    - [ ] add inner x/y axis (inside each cell group) that shows only grid 
    - [ ] decide which properties get applied to the outer axis (e.g., ticks)

### PR DESCRIPTION
**Pull request by kanitw**
- Drawing grid for facet inside each cell and draw border around each cell.  Fix #920 
vega_editor

- Add extra rule mark to the manually drawn grid to close the table when faceting without inner x/y axis. Fix #803
vega_editor

- always add offset to facet's axis
- Don't use band of row/column scales to make sure the border of the cell works correctly

### CODE REVIEW

**Comment by kanitw** [comment]
- Drawing grid for facet inside each cell and draw border around each cell.  Fix #920 
vega_editor

- Add extra rule mark to the manually drawn grid to close the table when faceting without inner x/y axis. Fix #803
vega_editor

**Comment by domoritz** [comment]
Sweeeet

**Comment by domoritz** [comment]
The thick line issue is gone with https://github.com/vega/vega/pull/521. No need to add offset to background.

File: src/compile/axis.ts [hunk 1]
```diff
@@ -7,6 +7,47 @@ import {formatMixins} from './util';
 // https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#11-ambient-declarations
 declare let exports;
 
+/**
+ * Make an inner axis for showing grid for shared axis.  
+ */
+export function compilerInnerAxis(channel: Channel, model: Model) {
```
Comment (domoritz) [inline src/compile/axis.ts]: compile or compiler?

**Comment by domoritz** [comment]
There is something funny things going on with the offsets. Look at the examples and change row to column.

screen shot 2016-02-11 at 22 33 25
screen shot 2016-02-11 at 22 33 51
screen shot 2016-02-11 at 22 34 17

**Comment by kanitw** [comment]
> Also it seems odd to me that there is space between the axis and the grid only for the y axis.

See my note in #948 about vega/vega#519.  You're welcome to fix that.  Shouldn't be too hard.

**Comment by kanitw** [comment]
Re: stacked bar.  Great that you're finding this bug.   I need to think about how to solve it.  One thing that's terrible about d3 ordinal is that padding is a ratio, not integer.   ><"

**Comment by kanitw** [comment]
Ok. The other issue fixed in https://github.com/vega/vega-lite/commit/4a9185885d6617d6ad6258935ca7fc5dc681e465

**Comment by domoritz** [comment]
https://github.com/vega/vega/issues/519 has been fixed.  So we should be able to add offsets now.

**Pull request merged**